### DISCUSSION TITLE
Range of ABL mesh values shown in mesh viewer is not correct if all are positive

### DISCUSSION THREAD

**Discussion by Will-Hollingworth** [Jyers/Marlin#1304] [Q&A] [answered]
Using E3V2-BLTouch-3x3-HS-v4.2.2-v1.3.5 and the resulting mesh in my case has positive values for all 9 points. See the attached photo example. The mesh viewer shows the range as "Red -0.060.. ". I now realize that it seems to be picking a range for the red that is the -abs of the minimum mesh value - but it was confusing at first because there wasn't a negative point and nothing is shown in red.

On that note, why does the center point always seem to have an offset (0.06 in my case)? I assume it's the difference in height of the initial center measurement vs the center measurement from measuring the mesh. If that is the case then why is it so large and always repeatable? If it's really the same point shouldn't it be zero +/- the BL Touch repeatability tolerance? 

![IMG_3591(1) (Medium)](https://user-images.githubusercontent.com/26444228/129274775-a6a789e3-2846-446c-8eea-f1455a805e59.jpg)

**Comment by Jyers** [comment, marked as answer]
Don't forget the mesh "adapts" to the bed. This means that that middle point isn't necessarily just the sensor reading at the middle, but the inferred average value of that region calculated based on the surrounding areas
